### ISSUE TITLE
build never finishes (7.12.0)

### ISSUE THREAD

**Issue by laurelnaiad** [TypeStrong/atom-typescript#745]
I had problems like this in 7.11.2, but I attributed them to the "Progress events" problem that I was also having (and which seems to be somewhat ameliorated now).

I'm still not getting complete builds, though.
![screen shot 2015-11-25 at 12 26 22 pm](https://cloud.githubusercontent.com/assets/4078946/11408746/f4692826-936f-11e5-9b14-97b1d23bb783.png)

The relevant tsconfig.json (unless it's finding some in the .Trash as well!) is

```
{
    "buildOnSave": false,
    "compileOnSave": true,
    "indentSize": 2,
    "tabSize": 2,
    "compilerOptions": {
        "target": "es6",
        "module": "commonjs",
        "experimentalDecorators": true,
        "emitDecoratorMetadata": true,
        "declaration": false,
        "noImplicitAny": false,
        "removeComments": false,
        "noLib": false,
        "preserveConstEnums": true,
        "suppressImplicitAnyIndexErrors": true,
        "inlineSources": true,
        "inlineSourceMap": true
    },
    "filesGlob": [
        "./typings/tsd.d.ts",
        "./global.d.ts",
        "./**/*.ts",
        "../gulpfile.js",
        "!**/bak/**"
    ],
    "files": [ "<omitted while pasting>" ],
    ],
    "atom": {
        "rewriteTsconfig": true
    }
}
```

I'd be interested in fixing this myself, or at least hunting it down, but I haven't yet grok'ed  where in the codebase the build process is managed.

The net results is that I can't do a build. Once it gets into this state (after the first build I try), it won't even try to do another unless I reload the atom view (in which case it will hang at this point the next time).

**Comment by laurelnaiad** [comment]
I figured out why my build never completes.

In the dev console, I saw there was an error that broke the promise chain to the point where a failure was lost. 

In this particular case, it seems I didn't have `"allowNonTsExtensions": true,` set, while also having a .js file in the build. This was the result of me futzing with the config while trying to figure out why babel wasn't working (now solved in #744, but with my tsconfig.json left in a state I never intended it to be).

So, the error was

```
Error: Could not find file: '/Users/dmaddox/Documents/_ws/proj/gulpfile.js'.
    at getValidSourceFile (/Users/dmaddox/Documents/_ws/atom-typescript/node_modules/ntypescript/bin/ntypescript.js:46942:23)
    at Object.getEmitOutput (/Users/dmaddox/Documents/_ws/atom-typescript/node_modules/ntypescript/bin/ntypescript.js:49926:30)
    at Object.emitFile (/Users/dmaddox/Documents/_ws/atom-typescript/dist/main/lang/modules/building.js:25:27)
    at /Users/dmaddox/Documents/_ws/atom-typescript/dist/main/lang/projectService.js:55:31
    at Array.map (native)
    at Object.build (/Users/dmaddox/Documents/_ws/atom-typescript/dist/main/lang/projectService.js:54:31)
    at Child.RequesterResponder.processRequest (/Users/dmaddox/Documents/_ws/atom-typescript/dist/worker/lib/workerLib.js:48:60)
    at process.<anonymous> (/Users/dmaddox/Documents/_ws/atom-typescript/dist/worker/lib/workerLib.js:226:23)
    at emitTwo (events.js:87:13)
    at process.emit (events.js:172:7)
```

I think there are really two problems.

1. unhandled errors in the build shouldn't be lost! :)
2. including .js files while not allowing nonTS extensions is a problem that ought not be caught only at the point where it throws an unhandled error.

I have a patch which catches otherwise unhandled errors which happen in `::getEmitOutput` and brute-forces an error into the results.

I don't much like my patch -- it's kind of a spot fix for when `services.getEmitOutput` throws, and the CodeError created in line is probably not the "right" way to report it.

I'll do a PR anyway. I don't yet know the codebase well enough to do it right quickly, but dropping build errors is probably not something that's desirable, so the patch might be helpful to have out there in the meantime. Perhaps someone will have a better solution to offer in the context of the PR.

**Comment by basarat** [comment]
Thanks for tracking it down. Created a clear issue up for future : https://github.com/TypeStrong/atom-typescript/pull/747 :rose:
